src/System.Private.ServiceModel/tools/scripts/StopWcfSelfHostedSvcIfRunning.cmd

```diff
@@ -0,0 +1,67 @@
+@if "%_echo%" neq "on" echo off
+setlocal
+
+REM This CMD stops the Self hosted WCF test service.
+REM It will self-elevate to kill the local process.
+
+set _setupMonitorFile=%~dp0..\..\..\..\SelfHostedWcfServiceMonitor.log
+
+REM Detect if the process is running locally
+tasklist /FI "imagename eq SelfHostedWCFService.exe" | findstr /i /C:"SelfHostedWCFService.exe" 1>nul
+if [%ERRORLEVEL%] == [0] goto stopService
+ 
+REM If it's not running, exit with 0 to indicate nothing was done  
+echo Detected Self hosted WCF Test Service is not running.
+exit /b 0
+
+:stopService
+
+REM We will open a separate elevated window to do clean up only if the current window is not elevated
+REM This is to improve the debugging experience
+net session >nul 2>&1
+if NOT [%ERRORLEVEL%]==[0] (
+    set _runelevated=%~dp0RunElevated.vbs
+)
+
+echo Stopping SelfHostedWCFService.exe process.
+
+REM Note: when elevating, we don't redirect output to the _cleanuplog
+REM due to contention for the log file.  Normal use is already elevated
+REM and captures everything to the log file
+if '%_runelevated%' == '' (
+  call %~dp0StopWcfSelfHostedProcess.cmd
+) else (
+  call %_runelevated% %~dp0StopWcfSelfHostedProcess.cmd >nul
+)
+
+if NOT [%ERRORLEVEL%]==[0] (
+	echo Warning: An error occurred while killing the SelfHostedWCFService.exe.
+	exit /b 1
+	)
+	
+REM Terminating the Self host WCF process started with StartSelfHostWcfSvc.cmd
+REM results in that script gaining control to finish cleanup.  If we are in that
+REM state, we need to wait for that cleanup to finish, or the bin folder will still
+REM be in use.
+
+for /l %%x in (1, 1, 60) do (
+  echo Waiting for Self hosted WCF service cleanup to complete...
+  if NOT EXIST %_setupMonitorFile% (
+    echo Self hosted WCF service cleanup has completed.
+    exit /b -1
+  )
+  
+  REM The "proper" way to wait is 'timeout' but that unconditionally
+  REM exits the process with "Input redirection is not supported".
+  REM The community preferred approach to wait is this.
+  ping 127.0.0.1 -n 6 > nul
+
+echo %%x
+
+)
+
+echo Failed to finish cleanup for Self hosted WCF service
+if EXIST %_setupMonitorFile% del %_setupMonitorFile%
+exit /b -1
```

Comment:
What does -1 mean? Who is going to use it? Should it be documented somewhere if -1 is necessary?